src/coordinator/parser/promql/types.go

```diff
@@ -82,12 +83,12 @@ func NewBinaryOperator(expr *promql.BinaryExpr, lhs, rhs parser.NodeID) (parser.
 // NewFunctionExpr creates a new function expr based on the type
 func NewFunctionExpr(name string, argValues []interface{}) (parser.Params, error) {
 	switch name {
-	case functions.AbsType:
-		return functions.AbsOp{}, nil
-	case functions.ClampMinType:
-		return functions.NewClampOp(argValues, functions.ClampMinType)
-	case functions.ClampMaxType:
-		return functions.NewClampOp(argValues, functions.ClampMaxType)
+	case linear.AbsType:
+		return linear.NewAbsOp(argValues), nil
```

Comment:
Maybe check in the `NewAbsOp` function that `len(args) == 0`?

Reply:
that's probably the promql's internal parsers responsibility. I don't think we care